replica/table.cc

```diff
@@ -3549,17 +3549,6 @@ future<> table::cleanup_tablet(database& db, db::system_keyspace& sys_ks, locato
     _sg_manager->mark_storage_group_for_remove(tid.value());
 }
 
-future<> table::cleanup_tablet_without_deallocation(database& db, db::system_keyspace& sys_ks, locator::tablet_id tid) {
-    auto holder = async_gate().hold();
-    // Hold shared lock to keep storage group alive.
-    rwlock::holder shared_lock = co_await _sg_manager->get_rwlock().hold_read_lock();
-    auto* sg = storage_group_for_id(tid.value());
-
-    co_await clear_inactive_reads_for_tablet(db, sg);
-    co_await flush_compaction_groups(sg);
```

Comment:
This is different than `table::cleanup_tablet`.
We only flush the compaction groups, not stop them.

Reply:
Right, I guess this is the reason for the test failures.